microprofile/config/config-cdi/src/main/java/io/helidon/microprofile/config/cdi/ConfigCdiExtension.java

```diff
@@ -131,8 +138,49 @@ any java class (except for parametrized types)
 
             pip.configureInjectionPoint()
                     .addQualifier(q);
+        }
+    }
+
+    private void harvestConfigPropertyInjectionPointsFromEnabledBean(@Observes ProcessBean<?> event) {
+        Bean<?> bean = event.getBean();
+        Set<InjectionPoint> beanInjectionPoints = bean.getInjectionPoints();
+        if (beanInjectionPoints != null && !beanInjectionPoints.isEmpty()) {
```

Comment:
Mostly to satisfy my own curiosity... There are several null-checks in this method. The JavaDoc for the invoked methods does not say that null might be returned (although it doesn't say null _will not_ ever be returned either although one could infer that). Are you just being really careful or have you seen these methods return null in your past experience with CDI?

Reply:
Mostly being careful.  I've seen `NullPointerException`s from non-market-leader CDI implementations in odd places and it has scarred me.  Also, a `Bean` in particular is an interface that an end user may implement, so she may return `null` from a variety of its methods and the specification does not forbid her from doing so.  So yes, being careful.